software/contrib/euclid.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python3
+"""Euclidean rhythm generator for the EuroPi
+
+@author Chris Iverach-Brereton <ve4cib@gmail.com>
+@author Brian House
+@year   2011, 2023
+
+@license MIT
```

Comment:
The MIT license is pretty short and broad:

> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
>
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
>
> THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

As long as the one function from Brian House is flagged as being licensed under the MIT terms, and he's given the appropriate credit for his work, I don't think it's an issue if it's included in a CC license.  I'm not a lawyer, but given the lack of a lot of restrictions in the MIT license I don't see any issue.

If we were adding GPL code that would be a different story. But in this case I think it should be okay.  Happy to be corrected by someone who knows more about this subject than I do though.

Reply:
I've tidied up the docstrings so it's clearer exactly what portion of the script is MIT-licensed.